android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt

```diff
@@ -2544,6 +2544,171 @@ class GeckoEngineSessionTest {
         assertFalse(onExceptionCalled)
     }
 
+    @Test
+    fun `onProductUrlChange should call onSuccess callback for a valid GV response`() {
```

Comment:
nit: let's format the test titles in the GIVEN, WHEN, THEN style

Reply:
oh right, re-formatted the test names